google-cloud-core/src/test/java/com/google/cloud/PolicyV3Test.java

```diff
@@ -0,0 +1,279 @@
+  private static final String ALL_AUTH_USERS = "allAuthenticatedUsers";
+  private static final String USER = "user:abc@gmail.com";
+  private static final String SERVICE_ACCOUNT = "serviceAccount:service-account@gmail.com";
+  private static final String GROUP = "group:group@gmail.com";
+  private static final String DOMAIN = "domain:google.com";
+  private static final String VIEWER = "roles/viewer";
+  private static final String EDITOR = "roles/editor";
+  private static final String OWNER = "roles/owner";
+  private static final List<String> MEMBERS_LIST_1 =
+      ImmutableList.of(USER, SERVICE_ACCOUNT, ALL_USERS);
+  private static final List<String> MEMBERS_LIST_2 =
+      ImmutableList.of(ALL_AUTH_USERS, GROUP, DOMAIN);
+  private static final List<Binding> BINDINGS_NO_CONDITIONS =
+      ImmutableList.of(
+          Binding.newBuilder().setRole(VIEWER).setMembers(MEMBERS_LIST_1).build(),
+          Binding.newBuilder().setRole(EDITOR).setMembers(MEMBERS_LIST_2).build());
+  private static final List<Binding> BINDINGS_WITH_CONDITIONS =
+      ImmutableList.copyOf(BINDINGS_NO_CONDITIONS)
+          .of(
+              Binding.newBuilder()
+                  .setRole(VIEWER)
+                  .setMembers(MEMBERS_LIST_1)
+                  .setCondition(
+                      Condition.newBuilder()
+                          .setTitle("Condition")
+                          .setDescription("Condition")
+                          .setExpression("Expr")
+                          .build())
+                  .build(),
+              Binding.newBuilder().setRole(EDITOR).setMembers(MEMBERS_LIST_2).build());
+  private static final Policy FULL_POLICY_V1 =
+      Policy.newBuilder().setBindings(BINDINGS_NO_CONDITIONS).setEtag("etag").setVersion(1).build();
+
+  private static final Policy FULL_POLICY_V3 =
+      Policy.newBuilder()
+          .setBindings(BINDINGS_WITH_CONDITIONS)
+          .setEtag("etag")
+          .setVersion(3)
+          .build();
+
+  private static final Policy FULL_POLICY_V3_WITH_VERSION_1 =
+      Policy.newBuilder()
+          .setBindings(BINDINGS_WITH_CONDITIONS)
+          .setEtag("etag")
+          .setVersion(1)
+          .build();
+
+  @Test
+  public void testBuilder() {
+    assertEquals(BINDINGS_NO_CONDITIONS, FULL_POLICY_V1.getBindingsList());
+    assertEquals(1, FULL_POLICY_V1.getVersion());
+    assertEquals("etag", FULL_POLICY_V1.getEtag());
+    Policy policy = FULL_POLICY_V1.toBuilder().setBindings(BINDINGS_NO_CONDITIONS).build();
+    assertEquals(BINDINGS_NO_CONDITIONS, policy.getBindingsList());
+    assertEquals("etag", policy.getEtag());
+    assertEquals(1, policy.getVersion());
+
+    assertEquals(BINDINGS_WITH_CONDITIONS, FULL_POLICY_V3.getBindingsList());
+    assertEquals(3, FULL_POLICY_V3.getVersion());
+    assertEquals("etag", FULL_POLICY_V3.getEtag());
+    policy = FULL_POLICY_V3.toBuilder().setBindings(BINDINGS_WITH_CONDITIONS).build();
```

Comment:
Try breaking this up into multiple independent tests instead of reusing variables. That way they can fail independently without one failure shadowing the others.